src/helpers.cpp

```diff
@@ -111,6 +111,28 @@ std::vector<JointParameter> loadJointsFromParameter(
   return output;
 }
 
+std::unordered_map<std::string, double> loadGoalConstraintsFromParameter(
+      ros::NodeHandle& nodeHandle,
+      const std::vector<JointParameter>& jointParameters)
+{
+  std::unordered_map<std::string, double> goalConstraints;
+
+  for (auto& jointParam : jointParameters) {
+    std::string jointName = jointParam.mName;
+
+    double goalConstraint;
+    if (nodeHandle.getParam("constraints/" + jointName + "/goal", goalConstraint)) {
+      goalConstraints[jointName] = goalConstraint;
+    }
+  }
+  if (goalConstraints.size() == 0) {
```

Comment:
Nit: Please use `empty()`. `std::<container>::empty()` is generally faster or equal to `~::size() == 0` in performance.